discovery/azure/azure.go

```diff
@@ -629,3 +663,26 @@ func (client *azureClient) getNetworkInterfaceByID(ctx context.Context, networkI
 
 	return &result, nil
 }
+
+// addToCache will add the network interface information for the specified nicID
+// If the cache is disable we do nothing
+func (d *Discovery) addToCache(nicID string, netInt *network.Interface) {
+	if d.cache == nil {
+		return
+	}
+	rand.Seed(time.Now().UnixNano())
```

Comment:
why do you use random here?

Reply:
Because I didn't want the key to expire at the same time, which will cause the same effect as having no cache. 

If Key expire at the same time we will just consume all the api call that we try to avoid with the cache

Reply:
I don't,agree, if the cache lifetime is > refresh interval, the cache is worth it

Reply:
The idea of introducing a random bias to the expiration time to mitigate spikes when the initial batches of nodes expiring is quite reasonable.

Reply:
I have remove the seed put let the random [0-refreshInterval] added to the exp time.